graph/client/src/app/feature-tasks/tasks-sidebar.tsx

```diff
@@ -72,59 +83,54 @@ export function TasksSidebar() {
   }
 
   function selectProject(project: string) {
-    setSelectedProjects([...selectedProjects, project]);
-
-    const taskId = createTaskName(project, selectedTarget);
-
-    graphService.handleTaskEvent({
-      type: 'notifyTaskGraphTasksSelected',
-      taskIds: [taskId],
-    });
-  }
-
-  function selectAllProjects() {
-    navigate(
-      routeContructor(`/tasks/${encodeURIComponent(selectedTarget)}/all`, true)
-    );
-  }
-
-  function hideAllProjects() {
-    setSelectedProjects([]);
-
-    const allProjects = projects.map(
-      (project) => `${project.name}:${selectedTarget}`
-    );
-
-    graphService.handleTaskEvent({
-      type: 'notifyTaskGraphTasksDeselected',
-      taskIds: allProjects,
-    });
-
-    navigate(
-      routeContructor(`/tasks/${encodeURIComponent(selectedTarget)}`, true)
-    );
+    const newSelectedProjects = [...selectedProjects, project];
+    if (newSelectedProjects.length === allProjectsWithTarget.length) {
+      searchParams.delete('projects');
+      navigate({
+        pathname: `../${encodeURIComponent(selectedTarget)}/all`,
```

Comment:
The `..` route doesn't work as expected because you don't always have a target selected. When you first land on the task graph view, you'll just be at `/tasks`. The first target in the available dropdowns will be selected but not added to the url. If you click on 'Show all' while on `/tasks`, this will result in an invalid route. 

With the route constructor, you can do an absolute path which makes things a little easier: 
```
routeContructor({
        pathname: `/tasks/${encodeURIComponent(selectedTarget)}/all`,
        search: searchParams.toString()
      }, false)
```